src/main/kotlin/org/jetbrains/research/ml/ast/transformations/deadcode/DeadCodeRemovalTransformation.kt

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.research.ml.ast.transformations.deadcode
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.psi.PsiElement
+import com.jetbrains.python.psi.PyElementGenerator
+import com.jetbrains.python.psi.PyRecursiveElementVisitor
+import com.jetbrains.python.psi.PyStatementList
+import org.jetbrains.research.ml.ast.storage.MetaDataStorage
+import org.jetbrains.research.ml.ast.transformations.Transformation
+
+class DeadCodeRemovalTransformation(private val storage: MetaDataStorage) : Transformation {
+    override val metadataKey: String = "DeadCodeRemoval"
+
+    override fun apply(psiTree: PsiElement, toStoreMetadata: Boolean) {
+        val heuristicVisitor = DeadCodeRemovalHeuristicVisitor()
```

Comment:
Sorry, maybe I missed something, but why do you have such heuristics? Does it speed up the transformation? I just thought the CFG visitor handles these cases so there is no need to visit the tree twice